api/tools/src/main/java/org/pmiops/workbench/tools/FetchFireCloudUserProfile.java

```diff
@@ -63,6 +58,9 @@ public CommandLineRunner run(UserDao userDao, FireCloudService fireCloudService)
   }
 
   public static void main(String[] args) throws Exception {
-    new SpringApplicationBuilder(FetchFireCloudUserProfile.class).web(false).run(args);
+    new SpringApplicationBuilder(CommandLineToolConfig.class)
```

Comment:
Since only the child class ever changes, you could make a helper method that just takes in that class object.

But I think the simplest would be `runAouCommandLineApp(getClass(), args);`. You might be able to genericize out the first argument, but type erasure might get in the way.

The biggest win would be localizing all the Spring incantations to a generic main/wrapper class.

Reply:
It's not a clear win to me. Adding a layer of indirection has a price in terms of reader comprehension. This case is borderline, but since it's still a fairly simple builder invocation with only a few LOC, I'd rather leave it for now.

Reply:
I get the rationale. Like I said, it's borderline to me, so since you seem to feel strongly I'll make the change.

Reply:
Thanks.

I thought about it some more, and I realized I should've asked a different question first: How did we decide each CLI tool would be a separate spring application as opposed to having one app with different sub-commands that shared a Spring context.  Is Spring smart enough not to instantiate branches of the dependency tree it doesn't need. If not, that would be one argument for separating them.